tests/run-make/private-dependency/Makefile

```diff
@@ -0,0 +1,10 @@
+include ../tools.mk
+
+# ignore-cross-compile
+# only-linux
+# only-x86_64
+
+all:
+	$(RUSTC) --crate-type rlib foo.rs
```

Comment:
Since this is a new run-make test, can you please write it in Rust instead of adding new Makefiles?

Reply:
I don't think this test needs to use `run-make` at all, it should work fine as a UI test with aux-crates. There are some examples in #122665. `-L` flags can also use substitutions there.

Reply:
Thanks for your comment. I have updated it.